mod/layout/set/abstract/media.rb

```diff
@@ -0,0 +1,27 @@
+format :html do
+  def image_card
+    @image_card ||= card.fetch(trait: :image)
+  end
+
+  def image_src opts
+    return "" unless image_card
+    nest(image_card, view: :source, size: opts[:size])
+  end
+
+  def image_alt
+    image_card&.name
+  end
+
+  def text_with_image_args opts
+    opts.reverse_merge! title: _render_title, text: "", src: image_src(opts),
+                        alt: image_alt, size: :original
+  end
+
+  def text_with_image opts={}
+    @image_card = Card.cardish(opts[:image]) if opts[:image]
+    opts[:media_opts] = {} unless opts[:media_opts]
+    opts[:media_left_extras] = "" unless opts[:media_left_extras]
+    text_with_image_args opts
+    haml :media_snippet, opts
+  end
+end
```

Comment:
I did not entirely duplicate this code, I only added `media_left_extra` , I wasn't aware that it exists in decko as well. It would be nice if you could help to correct the changes that are arleady in Decko.

Reply:
I'm going to work on that.